File: src/commands/cache.ts
import {Arguments, Command} from "../Command.js";
import {downloadCacheDir} from "../Utils.js";
// @ts-ignore
import {formatSize} from "ktfile";

export default [
    new Command({
        force: Arguments.bool
    }, ["cache", "clean"], (flags) => {
        if (!flags.force) {
            throw "Please provide --force to confirm cache cleaning. Note that this action is irreversible.";
        }

        downloadCacheDir.listFiles().forEach(file => file.delete(true));
        printer.pass(`Cleared ${formatSize(downloadCacheDir.size)} MB of download cache.`);
    }),
    new Command({}, ["cache"], () => {
        printer.info(`Current download cache size: ${formatSize(downloadCacheDir.size)} MB`);
    })
];

File: crates/craft-cli/src/commands/cache.rs
use colored::Colorize;
use craft_core::{CraftError, CraftPaths, Result};
use craft_providers::CacheManager;
use crate::cli::CacheCommands;

pub fn handle_cache(action: Option<CacheCommands>, paths: &CraftPaths) -> Result<()> {
    let cache = CacheManager::new(paths);

    match action {
        None => {
            let size = cache.get_cache_size();
            let mb = (size as f64) / (1024.0 * 1024.0);
            println!("{}", format!("Current download cache size: {:.2} MB", mb).cyan());
        }
        Some(CacheCommands::Clean { force }) => {
            if !force {
                return Err(CraftError::Other(
                    "Please provide '--force' to confirm cache cleaning. This action is irreversible.".to_string(),
                ));
            }
            let cleaned = cache.clean_cache()?;
            let mb = (cleaned as f64) / (1024.0 * 1024.0);
            println!("{}", format!("Cleared {:.2} MB of download cache.", mb).green());
        }
    }

    Ok(())
}
